src/components/post-actions.tsx

```diff
@@ -0,0 +1,77 @@
+import { FC, useState } from "react";
+import { likePostAction, unlikePostAction } from "@/lib/actions/posts.actions";
+import {
+  CommentButton,
+  MumbleActions,
+  ReplyFilledIcon,
+  ReplyIcon,
+} from "@ost-cas-fea-25-26/pp-design-system";
+import Link from "next/link";
+
+type PostActionsProps = {
+  mumbleId: string;
+  comments: number;
+  likes: number;
+  liked: boolean;
+  deepLink: string;
+  isReply?: boolean;
+};
+
+type LikeStateProps = {
+  liked: boolean;
+  likes: number;
+};
+
+export const PostActions: FC<PostActionsProps> = ({
+  mumbleId,
+  comments,
+  likes,
+  liked,
+  deepLink,
+  isReply = false,
+}) => {
+  const [likedState, setLikedState] = useState<LikeStateProps>({
+    liked: liked,
+    likes: likes,
+  });
+
+  const handleLikeToggle = async (nextState: boolean) => {
+    if (nextState) {
+      await likePostAction(mumbleId);
+      setLikedState({ liked: true, likes: likedState.likes + 1 });
```

Comment:
maybe this would be a good case to use useReducer instead of useState, but it works, so I'm Ok if we leave it. 😊